src/UsgsAstroLsSensorModel.cpp

```diff
@@ -200,156 +200,120 @@ csm::ImageCoord UsgsAstroLsSensorModel::groundToImage(
+   double pixelPrec = desired_precision / approxLineRes * 0.9;
+
+   // Start bisection search for zero
+   for (int it = 0; it < 30; it++) {
+      double nextTime = ((firstTime * lastOffset) - (lastTime * firstOffset))
+                      / (lastOffset - firstOffset);
+      double nextOffset = computeViewingPixel(nextTime, ground_pt, adj).line - 0.5;
+      // We're looking for a zero, so check that either firstLine and middleLine have
+      // opposite signs, or middleLine and lastLine have opposite signs.
+      if ((firstOffset > 0) == (nextOffset < 0)) {
+         lastTime = nextTime;
+         lastOffset = nextOffset;
       }
-      else  if (0.0 == dz)
-      {
-         det = xPLine * yPSamp - xPSamp * yPLine;
-         dLine = (yPSamp * dx - xPSamp * dy);
-         dSamp = (xPLine * dy - yPLine * dx);
+      else {
+         firstTime = nextTime;
+         firstOffset = nextOffset;
       }
-      else
-      {
-         throw csm::Error(
-            csm::Error::ALGORITHM,
-            "Undefined case.",
-            "UsgsAstroLsSensorModel::groundToImage");
+      if (fabs(lastOffset - firstOffset) < pixelPrec) {
+         break;
       }
+   }
 
-      // Update line and sample estimates
-
-      if (0.0 == det)
-      {
-         throw csm::Error(
-            csm::Error::ALGORITHM,
-            "Divide by zero.",
-            "UsgsAstroLsSensorModel::groundToImage");
-      }
-      lineTemp += (dLine / det / DELTA_IMAGE);
-      sampleTemp += (dSamp / det / DELTA_IMAGE);
-
-      // Update ground delta
-
-      xSeed = x;
-      ySeed = y;
-      zSeed = z;
-      mode = -1;
-      imageToPlane(lineTemp, sampleTemp, height, adj,
-         xSeed, ySeed, zSeed, mode);
-      dx = x - xSeed;
-      dy = y - ySeed;
-      dz = z - zSeed;
-      len = dx * dx + dy * dy + dz * dz;
-
-   }  // while (preSquare < len)
+   // Check that the desired precision was met
+
+   double computedTime = ((firstTime * lastOffset) - (lastTime * firstOffset))
+                       / (lastOffset - firstOffset);
+   csm::ImageCoord calculatedPixel = computeViewingPixel(computedTime, ground_pt, adj);
+   // The computed viewing line is the detector line, so we need to convert that to image lines
+   auto referenceTimeIt = std::upper_bound(_data.m_IntTimeStartTimes.begin(),
+                                           _data.m_IntTimeStartTimes.end(),
+                                           computedTime);
+   if (referenceTimeIt != _data.m_IntTimeStartTimes.begin()) {
+      --referenceTimeIt;
+   }
+   size_t referenceIndex = std::distance(_data.m_IntTimeStartTimes.begin(), referenceTimeIt);
+   calculatedPixel.line += _data.m_IntTimeLines[referenceIndex] - 1
+                         + (computedTime - _data.m_IntTimeStartTimes[referenceIndex])
+                         / _data.m_IntTimes[referenceIndex];
+   csm::EcefCoord calculatedPoint = imageToGround(calculatedPixel, _data.m_RefElevation);

```

Comment:
Until #29 is merged in, this part may cause problems when the actual pixel location is close to 0 or 1 lines, because the imagetoground call here won't properly compute the ground point.  So, for HRSC the new ground point could be off by +-20 m. This will cause an error in the next section.